pkg/reconciler/inmemorychannel/controller/inmemorychannel_test.go

```diff
@@ -807,6 +922,43 @@ func makeUnknownDeployment() *appsv1.Deployment {
 	return d
 }
 
+func makeEventPolicy(eventPolicyName string) *v1alpha1.EventPolicy {
+	return &v1alpha1.EventPolicy{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: "eventing.knative.dev/v1alpha1",
+			Kind:       "EventPolicy",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: testNS,
+			Name:      eventPolicyName,
+		},
+		Spec: v1alpha1.EventPolicySpec{
+			To: []v1alpha1.EventPolicySpecTo{
+				{
+					Ref: &v1alpha1.EventPolicyToReference{
+						APIVersion: v1.SchemeGroupVersion.String(),
+						Kind:       "InMemoryChannel",
+						Name:       imcName,
+					},
+				},
+			},
+		},
+		Status: v1alpha1.EventPolicyStatus{},
+	}
+}
+
+func makeReadyEventPolicy() *v1alpha1.EventPolicy {
+	policy := makeEventPolicy(readyEventPolicyName)
+	policy.Status.Conditions = []apis.Condition{{Type: v1alpha1.EventPolicyConditionReady, Status: corev1.ConditionTrue}}
+	return policy
+}
+
+func makeUnreadyEventPolicy() *v1alpha1.EventPolicy {
+	policy := makeEventPolicy(unreadyEventPolicyName)
+	policy.Status.Conditions = []apis.Condition{{Type: v1alpha1.EventPolicyConditionReady, Status: corev1.ConditionFalse}}
+	return policy
+}
```

Comment:
The general pattern for shared resources is to have a shared package and functions with options in `pkg/reconciler/testing` why using this approach here?

Reply:
updated